ts/features/idpay/barcode/screens/IdPayBarcodeResultScreen.tsx

```diff
@@ -0,0 +1,232 @@
+        <Body>
+          {I18n.t("idpay.barCode.resultScreen.success.body", {
+            initiativeName: barcode.initiativeId
+          })}
+        </Body>
+        <VSpacer size={24} />
+        <View style={styles.barcodeContainer}>
+          <Barcode format="CODE128" value={trx} />
+          <H3 style={{ alignSelf: "center" }}>{trx}</H3>
+          <VSpacer size={32} />
+          <BarcodeExpireSlider
+            secondsExpirationTotal={(barcode.trxExpirationMinutes ?? 0) * 60}
+            secondsToExpiration={secondsTillExpire}
+          />
+        </View>
+      </GradientScrollView>
+    </>
+  );
+};
+
+const LoadingScreen = () => (
+  <SafeAreaView style={styles.loadingWrapper}>
+    <LoadingIndicator size={48} color="aqua" />
+    <VSpacer size={24} />
+    <H3>{I18n.t("idpay.barCode.resultScreen.loading.body")}</H3>
+    <VSpacer size={8} />
+    <Body>{I18n.t("idpay.barCode.resultScreen.loading.subtitle")}</Body>
+  </SafeAreaView>
+);
+
+const ErrorContent = ({ onCtaPress }: { onCtaPress: () => void }) => (
+  <OperationResultScreenContent
+    title={I18n.t("idpay.barCode.resultScreen.error.generic.body")}
+    action={{
+      label: I18n.t("global.buttons.close"),
+      accessibilityLabel: I18n.t("global.buttons.close"),
+      onPress: onCtaPress
+    }}
+    pictogram="umbrellaNew"
+  />
+);
+
+// -------------------- components --------------------
+
+const BarcodeExpireSlider = ({
+  secondsToExpiration,
+  secondsExpirationTotal
+}: {
+  secondsToExpiration: number;
+  secondsExpirationTotal: number;
+}) => {
+  const [seconds, setSeconds] = React.useState(secondsToExpiration);
+  const isCodeExpired = seconds === 0;
+  React.useEffect(() => {
+    const timer = setInterval(() => {
+      setSeconds(currentSecs => currentSecs - 1);
+    }, 1000);
+    if (seconds <= 0) {
+      setSeconds(0);
+      clearInterval(timer);
+    }
+    return () => clearInterval(timer);
+    // possible over-engineering, but we actually
+    // only need to run zero checks once the code is expired,
+    // there is no reason to rerun this hook every second
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isCodeExpired]);
+
+  const formattedExpireMinutesString = isCodeExpired
+    ? ""
+    : new Date(seconds * 1000).toISOString().slice(14, 19);
+  return (
+    <View style={{ alignContent: "center" }}>
+      <ProgressBar progressPercentage={seconds / secondsExpirationTotal} />
+      <VSpacer size={8} />
+      <View style={styles.centeredRow}>
+        <LabelSmall weight="Regular" color="black">
+          {
+            I18n.t(
+              `idpay.barCode.resultScreen.success.${
+                isCodeExpired ? "expired" : "expiresIn"
+              }`
+            ) + " " /* added spacing to better format */
+          }
+        </LabelSmall>
+        <LabelSmall weight="SemiBold" color="black">
+          {formattedExpireMinutesString}
+        </LabelSmall>
+      </View>
+    </View>
+  );
+};
```

Comment:
I think this could go to its own file under `../components`. This should reduce the complexity of this file.  What do you think?

Reply:
I myself was 50/50 on this decision, since it would've added more imports and long, verbose names; so since the file was hovering the 250 lines mark I pulled a "eeh, why not?"  and kept it as-is.
That said, why not? we can surely try